### PR TITLE
Defer 'nc_time_axis' import, making dependency optional.

### LINKED ISSUE #2113: Make nc_time_axis an optional dependency
For a library with such a specific purpose, it is a surprise to find that `nc_time_axis` is now a first-class Iris dependency. At the very least `nc_time_axis` should be only a deferred load (i.e. only when necessary) within `iris.plot`, but even this will still mean that we cannot build the documentation without nc_time_axis. This clearly is not a desirable situation to be in.

An ideal solution to this would be that `nc_time_axis` behaves like `iris_grib`; that is that it is a plugin module for Iris that relies on Iris, rather than being a library that Iris relies on.

### PR DESCRIPTION
**Pull request by pp-mo**
Hopefully closes #2113

### CODE REVIEW

File: lib/iris/plot.py [hunk 1]
```diff
@@ -37,14 +37,14 @@
 import matplotlib.pyplot as plt
 import matplotlib.ticker as mpl_ticker
 from mpl_toolkits.axes_grid.anchored_artists import AnchoredText
-import nc_time_axis
 import netcdftime
```
Comment (dkillick) [inline lib/iris/plot.py]: You could do the same with this import too: it also was added by #2067.

**Comment by dkillick** [comment]
Thanks @pp-mo! Do we also want to remove nc_time_axis from the conda requirements file? Or is that one dependency removal too far?

File: lib/iris/plot.py [hunk 2]
```diff
@@ -412,6 +412,16 @@ def _fixup_dates(coord, values):
         if coord.units.calendar == 'gregorian':
             r = [datetime.datetime(*date) for date in dates]
         else:
+            try:
+                import nc_time_axis
+            except ImportError:
```
Comment (dkillick) [inline lib/iris/plot.py]: Of course, the problem with raising an error is that the failing examples will still fail, just with this error message instead...

**Comment by pp-mo** [comment]
> failing examples will still fail

I think I've sorted this out now...
As nc_time_axis is an _optional_ dependency, the tests should run without it.
They now do (and before was not absent from the 'minimal' test set).

Meanwhile, the examples can still require it, as they are only run with the full dependencies.

File: lib/iris/plot.py (hunk 1 above)
Comment (pp-mo) [inline lib/iris/plot.py]: But I think it's not worth it, as netcdf is a required dependency, ***not*** optional
(and it's just part of that)

Plus, it has nothing to do with the issue in hand.

File: lib/iris/plot.py [hunk 3]
```diff
@@ -412,6 +412,16 @@ def _fixup_dates(coord, values):
...
+                msg = ('Cannot plot against time in a non-gregorian '
+                       'calendar, because "nc_time_axis" is not available :  '
+                       'Install the package from '
+                       'https://github.com/SciTools/nc-time-axis to enable '
+                       'this usage.')
+                raise IrisError(msg)
```
Comment (dkillick) [inline lib/iris/plot.py]: What was the purpose of raising this as the slightly esoteric `IrisError`?

**Comment by marqh** [comment]
I think that this looks ready to merge.

The use of the IrisError makes sense to me, as Iris is unable to function in this case, due to a dependency problem.  Unless there's an established pattern for this kind of error, I'm content with this one

@pp-mo anything outstanding from your p.o.v.?

**Comment by marqh** [comment]
@pp-mo 
this is a 1.10 issue, please target the v1.10.x branch, rather than master?

**Comment by marqh** [comment]
Hi @pp-mo 
please may you remove the reference to nc_time_axis from 
`minimal-conda-requirements`
as part of this PR

thank you

**Comment by pp-mo** [comment]
> @marqh please may you remove the reference to nc_time_axis from minimal-conda-requirements
Some confusion here? : this was already done

**Comment by marqh** [comment]
> Some confusion here? : this was already done

so I see, my mistake
thank you

**Pull request merged**